pkg/asset/rhcos/image.go

```diff
@@ -94,20 +94,40 @@ func osImage(ctx context.Context, config *types.InstallConfig) (string, string,
 		if len(config.Platform.AWS.AMIID) > 0 {
 			return config.Platform.AWS.AMIID, "", nil
 		}
+		const globalResourceRegion = "us-east-1"
+		var osimageControlPlane, osimageCompute string
 		region := config.Platform.AWS.Region
-		if !rhcos.AMIRegions(config.ControlPlane.Architecture).Has(region) {
-			const globalResourceRegion = "us-east-1"
-			logrus.Debugf("No AMI found in %s. Using AMI from %s.", region, globalResourceRegion)
-			region = globalResourceRegion
+		{
+			if !rhcos.AMIRegions(config.ControlPlane.Architecture).Has(region) {
+				logrus.Debugf("No AMI found in %s. Using AMI from %s.", region, globalResourceRegion)
+				region = globalResourceRegion
+			}
+			osimageControlPlane, err = st.GetAMI(archName, region)
+			if err != nil {
+				return "", "", err
+			}
+			if region != config.Platform.AWS.Region {
+				osimageControlPlane = fmt.Sprintf("%s,%s", osimageControlPlane, region)
+			}
 		}
-		osimageControlPlane, err := st.GetAMI(archName, region)
-		if err != nil {
-			return "", "", err
+		if len(config.Compute) == 0 || config.Compute[0].Architecture == config.ControlPlane.Architecture {
+			return osimageControlPlane, "", nil
 		}
-		if region != config.Platform.AWS.Region {
-			osimageControlPlane = fmt.Sprintf("%s,%s", osimageControlPlane, region)
+		{
```

Comment:
Same as https://github.com/openshift/installer/pull/8698/commits/39e9578011c18bfea15fea4802fa6e136a6f61f4#diff-da71df8f5aaa245588aad9ad03eac91400cfa2afe2072f3eb04aa6125535e85dR108, unless the way the diffs are generated is tricking my eyes.

Reply:
The `{}` are just creating a scope so I can use `archName` without overriding the one from the outer scope. I can probably remove them now since I had a slightly different logic when I introduced them.

Reply:
Done